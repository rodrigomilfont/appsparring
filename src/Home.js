// @flow

import React from 'react';
// import logo from './logo.svg';

const Home = () => (
  <div className="Home">
    <header className="App-header">
      <h1 className="App-title">Welcome to React</h1>
      {/* <img src={logo} className="App-logo" alt="logo" /> */}
    </header>
    <h3>Welcome h3</h3>
    <h2>Welcome h2</h2>
    <h4>
      {' '}
      <a>Blah</a> {' '}
    </h4>
    <h5>H5</h5>
    <p className="App-intro">
      To get started, edit <code>src/.js</code> and save to reload.{' '}
    </p>{' '}
  </div>
);

export default Home;
