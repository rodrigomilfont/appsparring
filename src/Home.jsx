// @flow

import React from 'react';
import Header from './components/header/Header';

const Home = (props) => (
  <div className="grid">
    <Header {...props } />
    <aside className="sidebar-left">
      <article>
        <h4>
          <a href="#deded">Blah</a>
        </h4>
        <h2>Welcome h2</h2>
        <h5>Welcome H5</h5>
      </article>
    </aside>
    <main className="content">
      <article>
        <h3>Welcome h3</h3>
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>{' '}
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>{' '}
        <p>
          Is a non-negative <strong>percentage</strong> value, relative to the
          block size of the grid container. If the size of the grid container
          depends on the size of its tracks, then the percentage must be treated
          as auto. The intrinsic size contributions of the track may be adjusted
          to the size of the grid container, and increase the final size of the
          track by the minimum amount that would result in honoring the
          percentage.
        </p>
      </article>
    </main>
    <aside className="sidebar-rigth">
      <p>
        <span className="bt_nav">
          <a className="bt_home" href="#home">
            Home
          </a>
        </span>
      </p>
    </aside>
    <footer className="footer">
      <p className="footer-text">
        To get started, edit <code>src/.js</code> and save to reload.
      </p>
    </footer>
  </div>
);

export default Home;
