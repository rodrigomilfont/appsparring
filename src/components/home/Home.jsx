import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../header/Header';
import Menu from '../menu/Menu';
import donaGata from '../header/donaGata84.jpg';
import Felix from '../header/felix_the_cat.svg';
import './index.css';

const Home = props => (
  <div className="grid">
    <Header {...props} />
    <Menu />
    <main className="content">
      <article>
        <h1>Home</h1>
        <img src={donaGata} alt="" />
      </article>
      <img src={Felix} alt="Felix" width="200" />
    </main>
    <aside className="sidebar-rigth">
      <p>
        <span className="bt_nav">
          <Link to="/">Home Items</Link>
        </span>
      </p>
    </aside>
    <footer className="footer">
      <p className="footer-text">...</p>
    </footer>
  </div>
);

export default Home;
