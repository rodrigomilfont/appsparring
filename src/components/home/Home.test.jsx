import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Home from './Home';

Enzyme.configure({ adapter: new Adapter() });

it('renders Home', () => {
  const wrapper = shallow(<Home />);
  expect(wrapper.find('.footer').length).toEqual(1);
});

it('example of debug use', () => {
  const wrapper = shallow(<Home />);
  console.log(wrapper.find('.footer').debug());
});
