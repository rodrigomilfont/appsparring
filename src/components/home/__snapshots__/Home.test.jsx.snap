// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example of debug use 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          showSearch={false}
        />,
        <Menu />,
        <main
          className="content"
        >
          <article>
            <h1>
              Home
            </h1>
            <img
              alt=""
              src="test-file-stub"
            />
          </article>
          <img
            alt="Felix"
            src="test-file-stub"
            width="200"
          />
        </main>,
        <aside
          className="sidebar-rigth"
        >
          <p>
            <span
              className="bt_nav"
            >
              <Link
                replace={false}
                to="/"
              >
                Home Items
              </Link>
            </span>
          </p>
        </aside>,
        <footer
          className="footer"
        >
          <p
            className="footer-text"
          >
            ...
          </p>
        </footer>,
      ],
      "className": "grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "showSearch": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article>
              <h1>
                Home
              </h1>
              <img
                alt=""
                src="test-file-stub"
              />
            </article>,
            <img
              alt="Felix"
              src="test-file-stub"
              width="200"
            />,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Home
                </h1>,
                <img
                  alt=""
                  src="test-file-stub"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                },
                "ref": null,
                "rendered": "Home",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "src": "test-file-stub",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Felix",
              "src": "test-file-stub",
              "width": "200",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            <span
              className="bt_nav"
            >
              <Link
                replace={false}
                to="/"
              >
                Home Items
              </Link>
            </span>
          </p>,
          "className": "sidebar-rigth",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="bt_nav"
            >
              <Link
                replace={false}
                to="/"
              >
                Home Items
              </Link>
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Home Items
              </Link>,
              "className": "bt_nav",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home Items",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home Items",
              "type": [Function],
            },
            "type": "span",
          },
          "type": "p",
        },
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="footer-text"
          >
            ...
          </p>,
          "className": "footer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "...",
            "className": "footer-text",
          },
          "ref": null,
          "rendered": "...",
          "type": "p",
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            showSearch={false}
          />,
          <Menu />,
          <main
            className="content"
          >
            <article>
              <h1>
                Home
              </h1>
              <img
                alt=""
                src="test-file-stub"
              />
            </article>
            <img
              alt="Felix"
              src="test-file-stub"
              width="200"
            />
          </main>,
          <aside
            className="sidebar-rigth"
          >
            <p>
              <span
                className="bt_nav"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home Items
                </Link>
              </span>
            </p>
          </aside>,
          <footer
            className="footer"
          >
            <p
              className="footer-text"
            >
              ...
            </p>
          </footer>,
        ],
        "className": "grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "showSearch": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article>
                <h1>
                  Home
                </h1>
                <img
                  alt=""
                  src="test-file-stub"
                />
              </article>,
              <img
                alt="Felix"
                src="test-file-stub"
                width="200"
              />,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Home
                  </h1>,
                  <img
                    alt=""
                    src="test-file-stub"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "test-file-stub",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Felix",
                "src": "test-file-stub",
                "width": "200",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              <span
                className="bt_nav"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home Items
                </Link>
              </span>
            </p>,
            "className": "sidebar-rigth",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="bt_nav"
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Home Items
                </Link>
              </span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  Home Items
                </Link>,
                "className": "bt_nav",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home Items",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home Items",
                "type": [Function],
              },
              "type": "span",
            },
            "type": "p",
          },
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="footer-text"
            >
              ...
            </p>,
            "className": "footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "...",
              "className": "footer-text",
            },
            "ref": null,
            "rendered": "...",
            "type": "p",
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
